plugin/framework/cmds/install.go

```diff
@@ -0,0 +1,135 @@
+package cmds
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/safing/portmaster/plugin/shared"
+	"github.com/spf13/cobra"
+)
+
+type InstallCommandConfig struct {
+	PluginName   string
+	StaticConfig json.RawMessage
+	Types        []shared.PluginType
+	Privileged   bool
+}
+
+func InstallCommand(cfg *InstallCommandConfig) *cobra.Command {
+	var installDir string
+
+	cmd := &cobra.Command{
+		Use: "install",
+		Run: func(cmd *cobra.Command, args []string) {
+			// create the plugin directory
+			pluginDir := filepath.Join(installDir, "plugins")
+			if err := os.Mkdir(pluginDir, 0755); err != nil && !os.IsExist(err) {
+				log.Fatalf("failed to create plugin directory: %s", err)
+			}
+
+			// get the path to the executable
+			execPath, err := os.Executable()
+			if err != nil {
+				log.Fatalf("failed to determine executable path: %s", err)
+			}
+
+			// get the name of the plugin
+			pluginName := filepath.Base(execPath)
+			if cfg != nil && cfg.PluginName != "" {
+				pluginName = cfg.PluginName
+			}
+
+			pluginTarget := filepath.Join(pluginDir, pluginName)
+
+			// delete the plugin if it exists already
+			if err := os.Remove(pluginTarget); err != nil && !os.IsNotExist(err) {
+				log.Fatalf("failed to remove previous plugin version: %s", err)
+			}
+
+			// open our self
+			source, err := os.Open(execPath)
+			if err != nil {
+				log.Fatalf("failed to open plugin executable: %s", err)
```

Comment:
Does it make sense to continue execution of the function if `source` does not exist?